src/pages/index.js

```diff
@@ -38,54 +88,115 @@ const createCard = (cardData) => {
 const addCardObject = {
   popupSelector: "#add-card-modal",
   handleFormSubmit: (inputValues) => {
-    const cardElement = createCard(inputValues);
-    addCardForm.popupForm.reset();
-    addCardFormValidator.toggleButtonState();
-    cardSelection.addItem(cardElement);
-    addCardForm.close();
+    addCardForm.renderLoading(true, "Saving...");
+    api
+      .addCard(inputValues.name, inputValues.link)
+      .then((res) => {
+        // cardElement = createCard(inputValues);
+        // const cardElement = createCard(inputValues);
+        const cardElement = createCard(res);
+        // addCardForm.popupForm.reset();
+        // addCardFormValidator.toggleButtonState();
+        cardSelection.addItem(cardElement);
+      })
+      .then(() => {
+        addCardForm.renderLoading(false);
```

Comment:
`.renderLoading(false);` should be in block `finally`


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.